### PR TITLE
Omero CLI, Refactor R-omero installation

### PR DESCRIPTION
**Pull request by manics**
Install the OMERO CLI including support for imports. `omero` is in the default path so can be run without a prefix, e.g.
```
touch image.fake
omero login -s eel.openmicroscopy.org -u user-1
omero import image.fake
```
screen shot 2018-07-20 at 13 31 34
@pwalczysko 

- [x] --depends-on https://github.com/ome/rOMERO-gateway/pull/52

Related to https://trello.com/c/09QarUT5/37-basel-training-03-04-09-2018-setup

----
## Updated 2018-08-25

Additional changes:
- CellProfiler uses a conda JRE instead of a global one
- R is installed using a conda environment file
- OMERO Python2 Jupyter kernel logos have been changed from the OME logo to the OMERO python logo from https://www.openmicroscopy.org/omero/downloads/ (it'd be nice if we had an R one too!):
screen shot 2018-07-26 at 10 41 31

### CODE REVIEW

**Comment by jburel** [comment]
Maybe open a PR with only ac08c11 so we can merge the fix 
then we can review the R changes

**Comment by manics** [comment]
See https://github.com/ome/training-notebooks/pull/19

**Comment by jburel** [comment]
rgateway tagged to 0.4.1

**Comment by jburel** [comment]
ac08c11dca4ba371ecb5a24e2bb8e614dc25ecb2 has been merged
so it can be removed from this PR

File: Dockerfile [hunk 1]
```diff
@@ -4,98 +4,52 @@ MAINTAINER ome-devel@lists.openmicroscopy.org.uk
...
-    && apt-get -y install default-jdk
-
-ENV JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64
-
-# Cell Profiler
+# Cell Profiler (add to the Python2 environment)
 ADD docker/environment-python2-cellprofiler.yml .setup/
 RUN conda env update -n python2 -f .setup/environment-python2-cellprofiler.yml
+# CellProfiler has to be installed in a separate step because it requires
+# the JAVA_HOME environment variable set in the updated environment
```
Comment (jburel) [inline Dockerfile]: Having that step of the requirement file implies that now that we will have to copy it in each Docker file we want to have it installed.
Docker-cp, Docker-all
Maybe there is no way around it

File: docker/environment-r-omero.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+channels:
+- bioconda
+- conda-forge
+- defaults
+dependencies:
+- bioconductor-ebimage=4.13.*
+- maven=3.5.*
+- openjdk=8.0.*
```
Comment (jburel) [inline docker/environment-r-omero.yml:8]: openjdk is in both requirement files.
I don't know if it is possible to combine requirements files
saying that the apps might not have the same required version over time

**Comment by jburel** [comment]
R and CP notebooks work as expected
Trying to import from terminal
```
omero import image.fake
Using session c2608e57-b783-4514-b4e0-42c4971ae2d5 (trainer-1@outreach.openmicroscopy.org:4064). Idle timeout: 10 min. Current group: Lab1
Traceback (most recent call last):
  File "/opt/omero/OMERO.server/bin/omero", line 125, in <module>
    rv = omero.cli.argv()
  File "/opt/omero/OMERO.server/lib/python/omero/cli.py", line 1620, in argv
    cli.invoke(args[1:])
  File "/opt/omero/OMERO.server/lib/python/omero/cli.py", line 1095, in invoke
    stop = self.onecmd(line, previous_args)
  File "/opt/omero/OMERO.server/lib/python/omero/cli.py", line 1172, in onecmd
    self.execute(line, previous_args)
  File "/opt/omero/OMERO.server/lib/python/omero/cli.py", line 1254, in execute
    args.func(args)
  File "/opt/omero/OMERO.server/lib/python/omero/plugins/import.py", line 494, in importer
    self.do_import(command_args, xargs)
  File "/opt/omero/OMERO.server/lib/python/omero/plugins/import.py", line 505, in do_import
    stdout=out, stderr=err)
  File "/opt/omero/OMERO.server/lib/python/omero/java.py", line 140, in popen
    check_java(command)
  File "/opt/omero/OMERO.server/lib/python/omero/java.py", line 31, in check_java
    raise Exception("Java could not be found. (Executable=%s)" % command[0])
Exception: Java could not be found. (Executable=java)
jovyan@f9355320be5c:~$

```

File: Dockerfile [hunk 3]
```diff
@@ -4,98 +4,52 @@ MAINTAINER ome-devel@lists.openmicroscopy.org.uk
...
-
-USER jovyan
+ARG ROMERO_VERSION=v0.4.1
+RUN cd /opt/romero && \
+    curl https://raw.githubusercontent.com/ome/rOMERO-gateway/$ROMERO_VERSION/install.R --output install.R && \
+    bash -c "source activate r-omero && Rscript install.R --version=$ROMERO_VERSION"
+
+# OMERO full CLI
+# This currently uses the python2 environment, should we move it to its own?
+ARG OMERO_VERSION=5.4.6
```
Comment (mtbc) [inline Dockerfile]: I guess we can use 5.4.7 imminently. :smiley:

**Comment by dominikl** [comment]
Same here, notebooks (including R notebook) load fine, but same Java error in command line for 'omero' CLI.

**Comment by manics** [comment]
Sorry about that, I forgot to re-test OMERO CLI after the JDK work for CellProfiler and R. Should be working now:
screen shot 2018-07-26 at 10 26 01

I've also rebased to include the Travis fix, and I've updated the CLI to 5.4.7, the python-omero conda package is still at 5.4.6 though: https://github.com/ome/training-notebooks/blob/35ce8f1d13dc03e2d729e7550095cd23d9d3e555/docker/environment-python2-omero.yml#L6

File: Dockerfile (hunk 3 above)
Comment (manics) [reply Dockerfile]: Done!

File: docker/environment-r-omero.yml (hunk 2 above)
Comment (manics) [reply docker/environment-r-omero.yml:8]: As discussed, it's easier to keep them separate.

File: Dockerfile (hunk 1 above)
Comment (manics) [reply Dockerfile]: As discussed, although it's two steps it's balanced out by the overall reduction in complexity of the Dockerfile

**Comment by manics** [comment]
Updated to rOMERO 0.4.2

**Comment by dominikl** [comment]
Looks good, rOMERO 0.4.2 installed, R and Cellprofiler notebook can be loaded and the CLI import mentioned in the description works too! 👍

**Comment by jburel** [comment]
With ffc8484 we can now rebuild the image

**Comment by manics** [comment]
The final commit updates the python-omero bioconda to 5.4.7

**Pull request merged**